scripts/test-tls13-obsolete-curves.py

```diff
@@ -155,17 +157,181 @@ def main():
+        ext[ExtensionType.supported_versions] = SupportedVersionsExtension()\
+            .create([TLS_1_3_DRAFT, (3, 3)])
+        ext[ExtensionType.supported_groups] = SupportedGroupsExtension()\
+            .create(groups)
+        sig_algs = [SignatureScheme.rsa_pss_rsae_sha256,
+                    SignatureScheme.rsa_pss_pss_sha256,
+                    SignatureScheme.ecdsa_secp256r1_sha256]
+        ext[ExtensionType.signature_algorithms] = SignatureAlgorithmsExtension()\
+            .create(sig_algs)
+        ext[ExtensionType.signature_algorithms_cert] = SignatureAlgorithmsCertExtension()\
+            .create(SIG_ALL)
+        node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
+        node = node.add_child(ExpectAlert(AlertLevel.fatal,
+                                          alert_desc))
+
+        node = node.add_child(ExpectClose())
+        conversation_name = "secp256r1 and {0} in supported_groups and key_share".format(obsolete_group_name)
+        conversations[conversation_name] = conversation
+
+        # check inconsistent key_share with supported_groups
+        conversation = Connect(host, port)
+        node = conversation
+        ciphers = [CipherSuite.TLS_AES_128_GCM_SHA256,
+                   CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
+        ext = {}
+        groups = [GroupName.secp256r1]
+        key_shares = []
+        try:
+            key_shares.append(key_share_gen(obsolete_group))
+        except ValueError:
+            # bogus value to move on, if it makes problems, these won't result in handshake_failure
+            key_shares.append(KeyShareEntry()
+                .create(obsolete_group, bytearray(b'\xab'*32)))
+        ext[ExtensionType.key_share] = ClientKeyShareExtension().create(key_shares)
+        ext[ExtensionType.supported_versions] = SupportedVersionsExtension()\
+            .create([TLS_1_3_DRAFT, (3, 3)])
+        ext[ExtensionType.supported_groups] = SupportedGroupsExtension()\
+            .create(groups)
+        sig_algs = [SignatureScheme.rsa_pss_rsae_sha256,
+                    SignatureScheme.rsa_pss_pss_sha256,
+                    SignatureScheme.ecdsa_secp256r1_sha256]
         ext[ExtensionType.signature_algorithms] = SignatureAlgorithmsExtension()\
             .create(sig_algs)
         ext[ExtensionType.signature_algorithms_cert] = SignatureAlgorithmsCertExtension()\
-            .create(RSA_SIG_ALL)
+            .create(SIG_ALL)
         node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
         node = node.add_child(ExpectAlert(AlertLevel.fatal,
                                           alert_desc))
 
         node = node.add_child(ExpectClose())
-        conversation_name = "{0} in TLS 1.3".format(obsolete_group_name)
+        conversation_name = \
+            "{0} in key_share and secp256r1 in "\
+            "supported_groups (inconsistent extensions)"\
```

Comment:
given those new tests end up with the same consequence (receive alert + close connection), wouldn't it be possible to reuse a single test function with different configurations?

Reply:
there are slight differences for expected alert, but yes, except for key_share and supported_groups the extensions are the same for all connections. Will look into fixing it.